### PR TITLE
📖 Define details of "Communicate key dates to the community" 

### PR DESCRIPTION
**Pull request by furkatgofurov7**
**What this PR does / why we need it**:
Tries to better define Information/communication channels/stakeholders and remove TBDs'. Any suggestions to better define the above points are welcome

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of: #9104
Fixes: #9638 

/area release
/kind documentation

### CODE REVIEW

**Comment by furkatgofurov7** [comment]
@kubernetes-sigs/cluster-api-release-team 
/cc @killianmuldoon @sbueringer @CecileRobertMichon

File: docs/release/release-tasks.md [hunk 1]
```diff
@@ -383,24 +383,27 @@ Information can be distributed via:
 
 * `sig-cluster-lifecycle` mailing list
   * Note: The person sending out the email should ensure that they are first part of the mailing list. If the email is sent out is not received by the community, reach out to the maintainers to unblock and approve the email.
-* Slack
+* #cluster-api slack channel
```
Comment (CecileRobertMichon) [inline docs/release/release-tasks.md:386]:
```suggestion
* #cluster-api Slack channel
```

nit

File: docs/release/release-tasks.md [hunk 2]
```diff
@@ -383,24 +383,27 @@ Information can be distributed via:
...
 
-Relevant information includes: (TBD)
+Relevant information includes:
 
-* Beta, RC, GA release
+* Beta, RC, GA and patch release
 * Start of code freeze
 * Implementation progress
-* ...
+* KubeCon week notice (if applicable)
```
Comment (CecileRobertMichon) [inline docs/release/release-tasks.md:398]: what does this mean?

Comment (furkatgofurov7) [reply docs/release/release-tasks.md:398]: Probably not relevant, will remove it

**Comment by furkatgofurov7** [comment]
@CecileRobertMichon thanks for reviewing, updated it accordingly, PTAL.

File: docs/release/release-tasks.md [hunk 3]
```diff
@@ -383,24 +383,26 @@ Information can be distributed via:
 
 * `sig-cluster-lifecycle` mailing list
   * Note: The person sending out the email should ensure that they are first part of the mailing list. If the email is sent out is not received by the community, reach out to the maintainers to unblock and approve the email.
-* Slack
+* #cluster-api Slack channel
 * Office hours
+* Release Team meetings
 * Cluster API book
-* ...
+* Conference talks
```
Comment (killianmuldoon) [inline docs/release/release-tasks.md:390]: Let's remove this as I don't think it's the responsibility of the comms team

File: docs/release/release-tasks.md [hunk 4]
```diff
@@ -383,24 +383,26 @@ Information can be distributed via:
...
+* Github Issue (when communicating beta release to providers)
```
Comment (killianmuldoon) [inline docs/release/release-tasks.md:391]: Can we add a link here to clarify?

File: docs/release/release-tasks.md [hunk 5]
```diff
@@ -383,24 +383,26 @@ Information can be distributed via:
...
+* Release Team retrospective / pre-scheduled release cutting meetings
 
-Stakeholders are: (TBD)
+Stakeholders are:
 
 * End users of Cluster API
 * Contributors to core Cluster API
 * Provider implementers
-* ...
+* Downstream users (if applicable)
```
Comment (killianmuldoon) [inline docs/release/release-tasks.md:405]: Let's remove this as not part of the release team's scope.

File: docs/release/release-tasks.md [hunk 6]
```diff
@@ -383,24 +383,26 @@ Information can be distributed via:
...
 
-Relevant information includes: (TBD)
+Relevant information includes:
 
-* Beta, RC, GA release
+* Beta, RC, GA and patch release
 * Start of code freeze
 * Implementation progress
-* ...
+* Release Team retrospective / pre-scheduled release cutting meetings
```
Comment (killianmuldoon) [inline docs/release/release-tasks.md:398]: I don't think this needs to be part of the communication of a release.

File: docs/release/release-tasks.md [hunk 7]
```diff
@@ -383,24 +383,26 @@ Information can be distributed via:
...
+* Github Issue (when communicating beta release to providers)
 
-Relevant information includes: (TBD)
+Relevant information includes:
 
-* Beta, RC, GA release
+* Beta, RC, GA and patch release
 * Start of code freeze
 * Implementation progress
-* ...
```
Comment (killianmuldoon) [inline docs/release/release-tasks.md:396]: Add "Release delays and changes if applicable."

File: docs/release/release-tasks.md (hunk 6 above)
Comment (furkatgofurov7) [reply docs/release/release-tasks.md:398]: Agreed, will remove it

File: docs/release/release-tasks.md (hunk 4 above)
Comment (furkatgofurov7) [reply docs/release/release-tasks.md:391]: Will add a ref to https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/release/release-tasks.md#communicate-beta-to-providers, if that is what you meant?

Comment (furkatgofurov7) [reply docs/release/release-tasks.md]: Thanks, done

**Review by killianmuldoon — COMMENTED** [review]
/lgtm

Thanks!

**Comment by furkatgofurov7** [comment]
@CecileRobertMichon hey, sorry for the ping. If there are no outstanding comments to be addressed on this one, can we go ahead with this change?

**Comment by sbueringer** [comment]
/approve

Looks good to me. Let's open a follow-up PR if there are further findings.

**Pull request merged**